### PR TITLE
UI bundle previews

### PR DESCRIPTION
**Pull request by mlr**
This PR adds an Antora UI bundle preview build and a full Antora preview build that will both utilize whatever UI bundle changes exist on your Pull Request.

The reason I went with both a UI bundle preview, and a full Antora build preview is so we have the best of both worlds.  Without the UI bundle preview build it's harder to make UI bundle changes that also require content changes to test properly.  Without the Antora build preview you don't have the ability to see how your UI changes impact the real Antora build.  With both previews we get both for added confidence in our change.  

We will also receive any lint errors that happen as part of the UI bundle build or the Antora site build as comments on the PR.

### CODE REVIEW

**Comment by mlr** [comment]
UI bundle preview build successful! :white_check_mark:
Deploying preview to GitHub Pages.

**Comment by mlr** [comment]
Antora site build successful! :white_check_mark:
Deploying draft to GitHub Pages.

**Comment by mlr** [comment]
UI bundle preview build successful! :white_check_mark:
Deploying preview to GitHub Pages.

**Comment by mlr** [comment]
Antora site build successful! :white_check_mark:
Deploying draft to GitHub Pages.

File: gulp.d/tasks/build-preview-pages.js [hunk 1]
```diff
@@ -45,7 +45,17 @@ module.exports = (src, previewSrc, previewDest, sink = () => map()) => (done) =>
...
+
+            // The following has been customized to enable pull request preview builds on Github Pages.
+            // It differs from the Antora Default UI by allowing for an optional UI_ROOT_PATH_PREFIX.
+            // This environment variable is set during build to add a prefix to the {{ uiRootPath }} variable.
+            // This enables us to have preview builds that live in their own subdirectory on GitHub Pages.
+            let uiRootPath = path.join(siteRootPath, '_')
+            if (uiModel.env.UI_ROOT_PATH_PREFIX) {
+              uiRootPath = path.join('/', uiModel.env.UI_ROOT_PATH_PREFIX, uiRootPath)
+            }
+            uiModel.uiRootPath = uiRootPath
```
Comment (mlr) [inline gulp.d/tasks/build-preview-pages.js:57]: This is the only real deviation necessary from the default antora UI build task. In order to support having the preview build reside in a subdirectory, I needed a way to provide the path prefix at build time.

File: src/css/ds-vars.css [hunk 2]
```diff
@@ -1,6 +1,6 @@
 :root {
   --ds-common-black: #090909;
-  --ds-common-white: #fff;
+  --ds-common-white: #ff0;
```
Comment (mlr) [inline src/css/ds-vars.css:3]: I'll remove this before merging, but needed a way to test.

**Comment by mlr** [comment]
UI bundle preview build failure! :x:
> [20:18:52] Using gulpfile ~/work/docs-ui/docs-ui/gulpfile.js [20:18:52] Starting 'lint'... [20:18:52] Starting 'lint:css'... [20:18:52] Starting 'lint:js'... [20:18:54] /home/runner/work/docs-ui/docs-ui/src/js/08-toc-rail.js 5:56 error Extra semicolon semi ✖ 1 problem (1 error, 0 warnings) 1 error and 0 warnings potentially fixable with the `--fix` option. [20:18:54] 'lint:js' errored after 1.91 s [20:18:54] [31mESLintError[39m in plugin "[36mgulp-eslint[39m" Message: Failed with 1 error [20:18:54] 'lint' errored after 1.92 s

**Comment by mlr** [comment]
Antora site build failure! :x:
> [20:19:16] Using gulpfile ~/work/datastax-docs-site/datastax-docs-site/ui-bundle-preview/gulpfile.js [20:19:16] Starting 'bundle'... [20:19:16] Starting 'clean'... [20:19:16] Finished 'clean' after 12 ms [20:19:16] Starting 'lint:css'... [20:19:16] Starting 'lint:js'... [20:19:18] /home/runner/work/datastax-docs-site/datastax-docs-site/ui-bundle-preview/src/js/08-toc-rail.js 5:56 error Extra semicolon semi ✖ 1 problem (1 error, 0 warnings) 1 error and 0 warnings potentially fixable with the `--fix` option. [20:19:18] 'lint:js' errored after 1.97 s [20:19:18] [31mESLintError[39m in plugin "[36mgulp-eslint[39m" Message: Failed with 1 error [20:19:18] 'bundle' errored after 1.98 s

**Comment by mlr** [comment]
UI bundle preview build successful! :white_check_mark:
Deploying preview to GitHub Pages.

**Comment by mlr** [comment]
Antora site build successful! :white_check_mark:
Deploying draft to GitHub Pages.

File: src/css/ds-vars.css (hunk 2 above)
Comment (eric-schneider) [reply src/css/ds-vars.css:3]: My eyes 😵

Comment (mlr) [reply src/css/ds-vars.css:3]: :joy:

**Review by eric-schneider — APPROVED** [review]
Everything looks great as far as I can tell. Nice work! 

The only thing I would say is that it would be nice if the preview and draft build notices could be merged into a single comment so that the comment stream doesn't get overly polluted. But I recognize that's probably just a "nice to have" and not necessarily a blocker.

**Comment by mlr** [comment]
> Everything looks great as far as I can tell. Nice work!
> 
> The only thing I would say is that it would be nice if the preview and draft build notices were merged into a single comment so that the comment stream doesn't get overly polluted. But I recognize that's probably just a "nice to have" and not necessarily a blocker.

Agreed. I had considered this too, but due to the way the builds are happening in parallel (bundle preview happens on this repository, Antora preview happens on `datastax-docs-site`) it would have been overly complex to manage the sequence of the comment updates. For example, you would potentially get a success message for the bundle preview, followed by a link to the Antora preview. Separating them is a bit more noisy, but I think it eliminates confusion as to which link is for which build.

**Comment by mlr** [comment]
UI bundle preview build successful! :white_check_mark:
Deploying preview to GitHub Pages.

**Comment by mlr** [comment]
Antora site build successful! :white_check_mark:
Deploying draft to GitHub Pages.

**Pull request merged**